router/voteCallbacks.go

```diff
@@ -106,21 +106,31 @@ func (r *Router) getEnvelopeStatus(request routerRequest) {
 }
 
 func (r *Router) getEnvelope(request routerRequest) {
-	// check pid
-	if len(request.ProcessID) != types.ProcessIDsize {
-		r.sendError(request, "cannot get envelope status: (malformed processId)")
-		return
-	}
 	// check nullifier
 	if len(request.Nullifier) != types.VoteNullifierSize {
 		r.sendError(request, "cannot get envelope status: (malformed nullifier)")
 		return
 	}
-	envelope, err := r.vocapp.State.Envelope(request.ProcessID, request.Nullifier, true)
+	txRef, err := r.Scrutinizer.GetVoteReference(request.Nullifier)
 	if err != nil {
 		r.sendError(request, fmt.Sprintf("cannot get envelope: (%s)", err))
 		return
 	}
+	block := r.vocapp.Node.BlockStore().LoadBlock(txRef.BlockHeight)
```

Comment:
I think the `State` should have this methods abstracted.
+ state.GetBlockByHeight(height)
+ state.GetBlockByHash(hash)
+ state.GetTx(height, txIndex)

I'm not sure the Tendermint  BlockStore is safe, if not we should add a `sync.Mutex`

The Scrutinizer then would have a method for geting a specific envelope (already Unmarshaled)

+ scrutinizer.GetEnvelope(nullifier)

Reply:
It seems like the BlockStore is safe for this:
```go
type BlockStore struct {
	db dbm.DB

	// mtx guards access to the struct fields listed below it. We rely on the database to enforce
	// fine-grained concurrency control for its data, and thus this mutex does not apply to
	// database contents. The only reason for keeping these fields in the struct is that the data
	// can't efficiently be queried from the database since the key encoding we use is not
	// lexicographically ordered (see https://github.com/tendermint/tendermint/issues/4567).
	mtx    tmsync.RWMutex
	base   int64
	height int64
}
```